### PR TITLE
Rustfmt

### PR DESCRIPTION
**Pull request by Geobert**
Use `rustfmt` for code.

The only thing not rustfmt related is .gitignore to avoid pushing VSCode config files.

For some reason, rustfmt seems to do a "all-features" parse, so I needed to add empty files for modules that where feature gated and not present.

And it seems that rustfmt has issue with `#[path]` so added some `#[rustfmt::skip]` on them.

 - [x] I have read the [contribution guidelines][Contributing]

I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

### CODE REVIEW

**Comment by Geobert** [comment]
Uh, what are these `crossbeam` errors?

**Comment by HeroicKatora** [comment]
CI failures seems unrelated
* `actix` related crates fail on the error you've already seen.
* `rocket` fails on some panicking proc-macro? https://github.com/HeroicKatora/oxide-auth/pull/93/checks?check_run_id=841030551

File: oxide-auth/src/code_grant/refresh.rs [hunk 1]
```diff
@@ -239,31 +237,58 @@ impl<'req> Refresh<'req> {
...
-            },
-            (RefreshState::Authenticating { request, client, passdata: _, token }, Input::Authenticated) => {
+            }
+            (
+                RefreshState::Authenticating {
+                    request,
+                    client,
+                    passdata: _,
+                    token,
+                },
```
[the remark this answers is not in the document]
Comment (Geobert) [reply oxide-auth/src/code_grant/refresh.rs:251]: I'll add that in my todo list for the next commit then :)

**Comment by HeroicKatora** [comment]
@Geobert Regarding CI failure I think the `Cargo.lock` that you pushed is out of sync with the dependencies listed. You probably have updated `actix` in your private copies, that might interact weirdly with version resolution. No worries though, just fix it with the PR updating `oxide-auth-actix` :slightly_smiling_face:

**Comment by Geobert** [comment]
Ok I see, I'm not very good with Cargo.lock file ^^'

**Comment by HeroicKatora** [comment]
As a library it really shouldn't required a `Cargo.lock` file in the first place but my experience in terms of reproducibility with `actix` and `rocket`, in particular when they required nightly, was quite poor so I added it. We might be able to outright delete the file now that both work on stable?

**Comment by Geobert** [comment]
I though `rocket` was still on nightly, am I wrong?

**Comment by HeroicKatora** [comment]
It's going to be on stable [quite soon](https://github.com/SergioBenitez/Rocket/issues/19), when `proc_macro_hygiene` hits stable. That feature was stabilized in May so it should be in either the next version or the one after that, don't know if it was in time for beta or the exact timing.

**Pull request merged**